pkg/kubernetes/kube.go

```diff
@@ -866,3 +873,187 @@ func (kc *Client) PersistentVolExists(volName, expectedPhase string) error {
 	}
 	return nil
 }
+func (kc *Client) SecretDelete(secretName, namespace string) error {
+	return kc.SecretOperationFromEnvironmentVariable(OperationDelete, secretName, namespace, "")
+}
+
+func (kc *Client) SecretOperationFromEnvironmentVariable(operation, secretName, namespace, environmentVariable string) error {
+	var (
+		secretValue string
+		ok          bool
+	)
+	if err := kc.Validate(); err != nil {
+		return err
+	}
+	if operation != OperationDelete {
+		secretValue, ok = os.LookupEnv(environmentVariable)
+		if !ok {
+			return errors.Errorf("couldn't lookup environment variable '%s'", environmentVariable)
+		}
+	}
+	switch operation {
+	case OperationCreate, OperationSubmit:
+		secret := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: secretName,
+			},
+			Data: map[string][]byte{
+				environmentVariable: []byte(secretValue),
+			},
+		}
+		_, err := kc.KubeInterface.CoreV1().Secrets(namespace).Create(context.TODO(), secret, metav1.CreateOptions{})
+		if kerrors.IsAlreadyExists(err) {
+			log.Infof("secret '%s' already created", secretName)
+			return nil
+		}
+		return err
+	case OperationUpdate:
+		currentSecret, err := kc.KubeInterface.CoreV1().Secrets(namespace).Get(context.TODO(), secretName, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		secret := currentSecret.DeepCopy()
+		if len(secret.Data) == 0 {
+			secret.Data = map[string][]byte{}
+		}
+		secret.Data[environmentVariable] = []byte(secretValue)
+		_, err = kc.KubeInterface.CoreV1().Secrets(namespace).Update(context.TODO(), secret, metav1.UpdateOptions{})
+		return err
+	case OperationDelete:
+		err := kc.KubeInterface.CoreV1().Secrets(namespace).Delete(context.TODO(), secretName, metav1.DeleteOptions{})
+		if kerrors.IsNotFound(err) {
+			log.Infof("secret '%s' already deleted", secretName)
+			return nil
+		}
+		return err
+	default:
+		return fmt.Errorf("unsupported operation: '%s'", operation)
+	}
+}
+
+func (kc *Client) KubernetesClusterShouldBe(state string) error {
+	switch state {
```

Comment:
Should this call kc.Validate() here? This step will happen very soon after cluster creation if done in BDD